yt/python/yt/wrapper/query_commands.py

```diff
@@ -195,3 +204,303 @@ def format_time(time):
+        If `KeyboardInterrupt` occurs, aborts query, finalizes and re-raises `KeyboardInterrupt`.
+
+        :param bool print_progress: print progress
+        :param float timeout: timeout of query in msec. If `None`, wait indefinitely.
+        :param bool raise_if_all_results_are_errors: if `True`, raise `YtQueryFailedError` if all results are errors
+        even for a completed query.
+        """
+
+        query_poll_period = get_config(self.client)["query_tracker"]["poll_period"] / 1000.0
+        time_watcher = TimeWatcher(min_interval=query_poll_period / 10.0,
+                                   max_interval=query_poll_period,
+                                   slowdown_coef=0.2)
+        print_info = self.printer if print_progress else lambda state: None
+
+        def abort():
+            self.abort(message="query aborted by client interruption")
+            for state in self.get_state_monitor(TimeWatcher(1.0, 1.0, 0.0)):
+                print_info(state)
+
+        abort_on_sigint = get_config(self.client)["query_tracker"]["abort_on_sigint"]
+
+        with ExceptionCatcher(self.abort_exceptions, abort, enable=abort_on_sigint):
+            start_time = time()
+            for state in self.get_state_monitor(time_watcher):
+                print_info(state)
+                if timeout is not None and time() - start_time > timeout / 1000.0:
+                    raise TimeoutError("Timed out while waiting for query to finish")
+
+        error = self.get_error(return_error_if_all_results_are_errors=raise_if_all_results_are_errors)
+        if error is not None:
+            raise error
+
+
+def run_query(engine, query, settings=None, files=None, stage=None, annotations=None, access_control_objects=None, sync=True, client=None):
+    """Run query and track its progress (unless sync = false).
+
+    :param engine: one of "ql", "yql", "chyt", "spyt".
+    :type engine: str
+    :param query: text of a query
+    :type query: str
+    :param settings: a dictionary of settings
+    :type settings: dict or None
+    :param files: a YSON list of files, each of which is represented by a map with keys "name", "content", "type". Field "type" is one of "raw_inline_data", "url"
+    :type files: list or None
+    :param stage: query tracker stage, defaults to "production"
+    :type stage: str
+    :param annotations: a dictionary of annotations
+    :type annotations: dict or None
+    :param access_control_objects: list access control object names
+    :type access_control_objects: list or None
+    """
+
+    query_id = start_query(engine, query, settings=settings, files=files, stage=stage,
+                           annotations=annotations, access_control_objects=access_control_objects, client=client)
+    query = Query(query_id, engine, query_tracker_stage=stage, client=client)
+    if query.url:
+        logger.info("Query started: %s", query.url)
+    else:
+        logger.info("Query started: %s", query.id)
```

Comment:
Don't we want a method like `Query.debug_descriptor` ?

Reply:
Not sure about a one-off method, rewrote this is `query.url or query.id`, should be fine for this usage.